src/main/java/org/openrewrite/kotlin/tree/K.java

```diff
@@ -567,17 +568,50 @@ public <T extends J> T withType(@Nullable JavaType type) {
     @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
     @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
     @RequiredArgsConstructor
-    @Getter
-    @With
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    @Data
     final class Constructor implements K, Statement, TypedTree {
 
+        @Nullable
+        @NonFinal
+        transient WeakReference<K.Constructor.Padding> padding;
+
         @EqualsAndHashCode.Include
+        @With
         UUID id;
 
+        @With
         Markers markers;
+
+        @With
         J.MethodDeclaration methodDeclaration;
-        Space colon;
-        ConstructorInvocation constructorInvocation;
+
+        // A replacement of `colon` and `constructorInvocation`
+        JLeftPadded<ConstructorInvocation> invocation;
+
+        public ConstructorInvocation getInvocation() {
+            return invocation.getElement();
+        }
+
+        public K.Constructor withInvocation(ConstructorInvocation invocation) {
+            return getPadding().withInvocation(this.invocation.withElement(invocation));
+        }
+
+        // For backward compatibility, handle removed fields `colon` and `constructorInvocation` which has been relocated to `invocation`
+        // Todo, Remove when we feel good that kotlin LSTs have been rebuilt.
+        @Deprecated
+        @JsonCreator
+        public Constructor(UUID id,
+                           Markers markers,
+                           J.MethodDeclaration methodDeclaration,
+                           @JsonProperty("colon") Space colon,
+                           @JsonProperty("constructorInvocation") ConstructorInvocation constructorInvocation) {
```

Comment:
We should declare these two parameters as `@Nullable`. As this constructor is declared as `@JsonConstructor`, Jackson will also call it when deserializing a new LST, which doesn't have these two properties. I tested that with the following tests in `TreeSerializerTest`:
```java
    @Test
    void jsonCreatorTest() throws IOException {
        ObjectMapper mapper = TreeSerializer.builder().build().buildMapper(getClass().getClassLoader(), StreamReadConstraints.defaults());
        ByteArrayOutputStream out = new ByteArrayOutputStream();

        FooOld input = new FooOld("foo");
        mapper.writeValue(out, input);
        FooNew deserialized = mapper.readValue(out.toByteArray(), FooNew.class);
        System.out.println(deserialized);
    }

    @Test
    void jsonCreatorTest2() throws IOException {
        ObjectMapper mapper = TreeSerializer.builder().build().buildMapper(getClass().getClassLoader(), StreamReadConstraints.defaults());
        ByteArrayOutputStream out = new ByteArrayOutputStream();

        FooNew input = new FooNew(2);
        mapper.writeValue(out, input);
        FooNew deserialized = mapper.readValue(out.toByteArray(), FooNew.class);
        System.out.println(deserialized);
    }

    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
    @RequiredArgsConstructor
    @Data
    static class FooOld {
        String name;
    }

    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
    @RequiredArgsConstructor
    @Data
    static class FooNew {
        Integer age;
        @JsonCreator
        public FooNew(String name) {
            this.age = name.length();
        }
    }
```